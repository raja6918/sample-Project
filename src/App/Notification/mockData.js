export const mockData = {
  data: [
    {
      activeTime: '0:00:05',
      alertType: 'NOTICE',
      creationDateTime: '2020-08-12T07:57:15+0000',
      elapsedTime: '0:00:00',
      eventId: 'JOB1001',
      eventType: 'solver',
      jobId: 'JOB1001',
      jobOwner: 11,
      lastModified: '2020-08-12T07:57:15+0000',
      message: 'string',
      notificationType: 'solver',
      scenarioId: 2,
      scenarioName: 'Test',
      solverId: 1,
      solverName: 'Test',
      state: 'Queued',
      status: 'Done-success',
      viewStatus: 'SEEN',
    },
  ],
  endIndex: 20,
  notViewedCount: 20,
  startIndex: 0,
  totalCount: 40,
};
