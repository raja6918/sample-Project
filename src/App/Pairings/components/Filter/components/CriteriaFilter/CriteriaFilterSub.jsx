import React, { Component } from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import { Button, Checkbox, Input } from '@material-ui/core';
import { withStyles } from '@material-ui/core/styles';
import { Menu, MenuItem, ListItemIcon, ListItemText } from '@material-ui/core';
import ArrowForwardIosIcon from '@material-ui/icons/ArrowForwardIos';
import PerfectScrollbar from 'react-perfect-scrollbar';
import CloseIcon from '@material-ui/icons/Close';
import { perfectScrollConfig } from '../../../../../../utils/common';

import {
  InputSearch,
  SearchIcon,
  ClearButton,
  ClearIcon,
} from '../../../../../../components/FilterInput';

const StyledSubComponent = styled.div`
  width: 180px;
  background: #ffffff;
  border-radius: 3px;
  outline: none;
  white-space: nowrap;

  & > div:first-child {
    display: flex;
    justify-content: center;
    border-bottom: 1px solid rgba(151, 151, 151, 0.24);
  }

  & > div:nth-child(2) {
    & > div:first-child {
      max-height: ${props => props.popupHeight + 'px'};
      & > div {
        display: flex;
        align-items: flex-start;
        padding: 4px 12px;
        font-size: 14px;
        color: rgba(0, 0, 0, 0.67);
        cursor: pointer;
        & > div {
          padding: 4px;
          white-space: normal;
          user-select: none;
        }
      }
      & > div:hover {
        background: #f5f5f5;
      }
    }
  }

  & > div:nth-child(3) {
    display: flex;
    flex-direction: column;
    justify-content: space-evenly;
    align-items: center;
    height: 86px;
    border-top: 1px solid rgba(151, 151, 151, 0.24);
  }
`;

const AddButton = styled(Button)`
  width: 80px;
  height: 32px;
  border-radius: 2px;
  box-shadow: none;
  margin-top: 10px;
`;

const ClearAllButton = styled(Button)`
  text-transform: none;
  font-size: 12px;
  padding: 0;
  margin-bottom: 10px;

  &:hover {
    background-color: transparent;
  }
  & .MuiButton-startIcon {
    margin-right: 0px;
  }
`;

const StyledInputSearch = styled(InputSearch)`
  div {
    border: 1px solid rgba(0, 0, 0, 0.54);
  }
`;

const StyledMenu = withStyles({
  paper: {
    border: '1px solid #d3d4d5',
    boxShadow: '0 2px 4px 0 rgba(0, 0, 0, 0.2)',
  },
})(props => (
  <Menu
    MenuListProps={{ disablePadding: true }}
    elevation={0}
    getContentAnchorEl={null}
    anchorOrigin={{
      vertical: 'top',
      horizontal: 'right',
    }}
    transformOrigin={{
      vertical: 'top',
      horizontal: 'left',
    }}
    {...props}
  />
));

const StyledMenuItem = withStyles(theme => ({
  root: {
    '& .MuiSvgIcon-fontSizeSmall': {
      fontSize: '16px',
    },
    '& .MuiTypography-body1': {
      fontWeight: 'unset',
      fontSize: '14px',
      whiteSpace: 'normal',
    },
    '&:hover': {
      backgroundColor: '#F5F5F5 !important',
    },
  },
  selected: {
    backgroundColor: '#EBEBEB !important',
    fontWeight: '500 !important',
  },
}))(MenuItem);

class CriteriaFilterSub extends Component {
  constructor(props) {
    super(props);
    this.state = {
      filteredItems: [],
      selectedIds: [],
      anchorEl: null,
      filter: '',
      popupHeight: 253,
    };
    this.handleClickOutside = this.handleClickOutside.bind(this);
  }
  popupTimer = null;

  componentDidMount() {
    this.setPopupHeight();
    document.addEventListener('mousedown', this.handleClickOutside);
    window.addEventListener('resize', this.setPopupHeight);
  }

  componentWillUnmount() {
    document.removeEventListener('mousedown', this.handleClickOutside);
    window.removeEventListener('resize', this.setPopupHeight);
    clearTimeout(this.popupTimer);
    this.popupTimer = null;
  }

  setPopupHeight = () => {
    const windowHeight = document.documentElement.clientHeight;
    let popupHeight;
    switch (true) {
      case windowHeight >= 700:
        popupHeight = 253;
        break;
      case windowHeight < 700 && windowHeight >= 440:
        popupHeight = 200;
        break;
      case windowHeight < 440 && windowHeight >= 400:
        popupHeight = 150;
        break;
      case windowHeight < 400 && windowHeight >= 350:
        popupHeight = 100;
        break;
      case windowHeight < 350 && windowHeight >= 300:
        popupHeight = 75;
        break;
      case windowHeight < 300 && windowHeight >= 200:
        popupHeight = 30;
        break;
      case windowHeight < 200:
        popupHeight = 0;
        break;
      default:
        popupHeight = 253;
    }
    this.setState({ popupHeight });
  };

  handleClickOutside = event => {
    try {
      const mainMenu = document.getElementById('main-menu');
      const subMenu = document.querySelectorAll('.sub-menu');
      let subMenuContain = false;
      subMenu.forEach(elem => {
        if (elem.contains(event.target)) {
          subMenuContain = true;
        }
      });
      if (mainMenu.contains(event.target) || subMenuContain) {
        return;
      }
      this.handleMenuClose();
    } catch (error) {
      console.error(error);
    }
  };

  handleMenuClick = currentTarget => {
    const { id, handleSubMenuOpen } = this.props;
    this.setState({ anchorEl: currentTarget });
    this.handleClearFilter();
    handleSubMenuOpen(id);
  };

  handleMenuClose = () => {
    this.props.handleSubMenuClose();
  };

  handleAdd = e => {
    const {
      getfilteredCriteria,
      catElement,
      handleClose,
      handleSubMenuClose,
    } = this.props;
    getfilteredCriteria(this.state.filteredItems, catElement.categoryId);
    this.handleClearFilter();
    this.handleClearAll();
    handleSubMenuClose();
    handleClose();
    e.stopPropagation();
  };

  handleClearAll = () => {
    this.setState({ filteredItems: [], selectedIds: [] });
  };

  handleFilter = e => {
    this.setState({ filter: e.target.value });
  };

  handleClearFilter = () => {
    this.setState({ filter: '' });
  };

  toggleItems = (item, checked) => {
    if (checked) {
      this.setState(prevState => ({
        filteredItems: [...prevState.filteredItems, item],
        selectedIds: [...prevState.selectedIds, item.crId],
      }));
    } else {
      this.setState(prevState => ({
        filteredItems: prevState.filteredItems.filter(
          filteredItem => filteredItem.crId !== item.crId
        ),
        selectedIds: prevState.selectedIds.filter(id => id !== item.crId),
      }));
    }
  };

  handlePopupOnEnter = event => {
    const currentTarget = event.currentTarget;
    this.popupTimer = setTimeout(() => {
      this.handleMenuClick(currentTarget);
    }, 300);
  };

  handlePopupOnLeave = () => {
    clearTimeout(this.popupTimer);
    this.popupTimer = null;
  };

  render() {
    const { catElement, t, subMenuOpen, id } = this.props;
    const { selectedIds, filter, anchorEl, popupHeight } = this.state;

    const sortedCriteria = this.props.catElement.criteria.sort((a, b) => {
      const aValue = this.props.t(`FILTER.filterCriteria.criteria.${a.crKey}`);
      const bValue = this.props.t(`FILTER.filterCriteria.criteria.${b.crKey}`);
      if (aValue < bValue) {
        return -1;
      }
      if (aValue > bValue) {
        return 1;
      }
      return 0;
    });

    const localisedMenuItemName = t(
      `FILTER.filterCriteria.categories.${catElement.categoryKey}`
    );
    return (
      <div>
        <StyledMenuItem
          onClick={event => this.handleMenuClick(event.currentTarget)}
          onMouseEnter={this.handlePopupOnEnter}
          onMouseLeave={this.handlePopupOnLeave}
          selected={id === subMenuOpen}
          id="main-menu"
        >
          <ListItemText primary={localisedMenuItemName} />
          <ListItemIcon style={{ minWidth: '20px', color: '#3f3e3e' }}>
            <ArrowForwardIosIcon fontSize="small" />
          </ListItemIcon>
        </StyledMenuItem>
        <StyledMenu
          style={{ pointerEvents: 'none' }}
          id="customized-menu"
          anchorEl={anchorEl}
          keepMounted
          open={id === subMenuOpen}
        >
          <StyledSubComponent
            style={{ pointerEvents: 'auto' }}
            className="sub-menu"
            popupHeight={popupHeight}
          >
            <div>
              <StyledInputSearch>
                <Input
                  onKeyDown={e => e.stopPropagation()}
                  type="text"
                  disableUnderline={true}
                  startAdornment={<SearchIcon>search</SearchIcon>}
                  placeholder={t('SEARCHENGINE')}
                  onChange={this.handleFilter}
                  value={filter}
                  endAdornment={
                    filter && (
                      <ClearButton onClick={this.handleClearFilter}>
                        <ClearIcon>highlight_off</ClearIcon>
                      </ClearButton>
                    )
                  }
                />
              </StyledInputSearch>
            </div>
            <PerfectScrollbar option={perfectScrollConfig}>
              <div>
                {sortedCriteria.map(crElement => {
                  const localisedCriteriaName = t(
                    `FILTER.filterCriteria.criteria.${crElement.crKey}`
                  );
                  if (
                    localisedCriteriaName
                      .toLowerCase()
                      .includes(filter.toLowerCase()) ||
                    filter === ''
                  ) {
                    const isItemPresent = selectedIds.includes(crElement.crId);
                    return (
                      <div
                        key={crElement.crId}
                        onClick={() =>
                          this.toggleItems(crElement, !isItemPresent)
                        }
                        style={{
                          background: isItemPresent ? '#EBEBEB' : '',
                          fontWeight: isItemPresent ? 500 : '',
                        }}
                      >
                        <Checkbox checked={isItemPresent} />
                        <div>{localisedCriteriaName}</div>
                      </div>
                    );
                  }
                  return null;
                })}
              </div>
            </PerfectScrollbar>
            <div>
              <AddButton
                variant="contained"
                color="primary"
                onClick={this.handleAdd}
                disabled={selectedIds.length === 0}
              >
                {t('FILTER.pane.buttons.add')}
              </AddButton>
              <ClearAllButton
                disabled={selectedIds.length === 0}
                color="primary"
                size="small"
                startIcon={<CloseIcon />}
                disableRipple={true}
                disableFocusRipple={true}
                onClick={this.handleClearAll}
              >
                {t('FILTER.pane.buttons.clearAll')}
              </ClearAllButton>
            </div>
          </StyledSubComponent>
        </StyledMenu>
      </div>
    );
  }
}

CriteriaFilterSub.propTypes = {
  t: PropTypes.func.isRequired,
  id: PropTypes.number.isRequired,
  getfilteredCriteria: PropTypes.func.isRequired,
  handleClose: PropTypes.func.isRequired,
  catElement: PropTypes.shape([]).isRequired,
  subMenuOpen: PropTypes.number,
  handleSubMenuOpen: PropTypes.func.isRequired,
  handleSubMenuClose: PropTypes.func.isRequired,
};

CriteriaFilterSub.defaultProps = {
  subMenuOpen: null,
};

export default CriteriaFilterSub;
