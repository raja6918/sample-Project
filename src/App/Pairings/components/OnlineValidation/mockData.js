export const alerts = {
  74: {
    alertLevel: 'error',
    alerts: {
      errorAlerts: [
        {
          rule: null,
          message: 'Invalid flight reference',
          tag: 'at',
          leg: {
            flightNumber: 8992,
            departureStationCode: 'YHZ',
            arrivalStationCode: 'YYT',
            startDate: '2019-07-08',
          },
          contextStartDateTime: '2019-07-08T19:45',
          contextEndDateTime: '2019-07-08T20:35',
        },
        {
          rule: null,
          message: 'Mismatched arrival and departure stations between flights.',
          tag: 'at',
          leg: {
            flightNumber: 8899,
            departureStationCode: 'YYR',
            arrivalStationCode: 'YHZ',
            startDate: '2019-07-08',
          },
          contextStartDateTime: '2019-07-08T19:45',
          contextEndDateTime: '2019-07-08T20:35',
        },
      ],
      warningAlerts: [
        {
          rule: { code: 'duty-credit', name: 'Duty credit' },
          message: 'duty credit exceeded',
          tag: 'at',
          leg: {
            flightNumber: 8829,
            departureStationCode: 'YYT',
            arrivalStationCode: 'YYR',
            startDate: '2019-07-08',
          },
          contextStartDateTime: '2019-07-08T19:45',
          contextEndDateTime: '2019-07-08T20:35',
        },
      ],
      cautionAlerts: [
        {
          rule: { code: 'leg-nb-max', name: 'Consecutive Flight Duty Periods' },
          message: 'Should be maximum of 3 but currently is 5',
          tag: 'at',
          leg: {
            flightNumber: 8885,
            departureStationCode: 'YYT',
            arrivalStationCode: 'YYR',
            startDate: '2019-07-08',
          },
          contextStartDateTime: '2019-07-08T19:45',
          contextEndDateTime: '2019-07-08T20:35',
        },
      ],
      infoAlerts: [
        {
          rule: { code: 'leg-nb-max', name: 'Consecutive Flight Duty Periods' },
          message: 'Should be maximum of 3 but currently is 5',
          tag: 'at',
          leg: {
            flightNumber: 8634,
            departureStationCode: 'YYT',
            arrivalStationCode: 'YYR',
            startDate: '2019-07-08',
          },
          contextStartDateTime: '2019-07-08T19:45',
          contextEndDateTime: '2019-07-08T20:35',
        },
      ],
    },
  },
  115: {
    alertLevel: 'warning',
    alerts: {
      warningAlerts: [
        {
          rule: { code: 'leg-nb-max', name: 'Maximum legs per duty' },
          message: 'maximum legs per duty exceeded',
          tag: 'at',
          leg: {
            flightNumber: 7951,
            departureStationCode: 'YYT',
            arrivalStationCode: 'YYR',
            startDate: '2019-07-08',
          },
          contextStartDateTime: '2019-07-08T19:45',
          contextEndDateTime: '2019-07-08T20:35',
        },
        {
          rule: { code: 'duty-credit', name: 'Duty credit' },
          message: 'duty credit exceeded',
          tag: 'at',
          leg: {
            flightNumber: 7954,
            departureStationCode: 'YYT',
            arrivalStationCode: 'YYR',
            startDate: '2019-07-08',
          },
          contextStartDateTime: '2019-07-08T19:45',
          contextEndDateTime: '2019-07-08T20:35',
        },
      ],
    },
  },
  116: {
    alertLevel: 'caution',
    alerts: {
      cautionAlerts: [
        {
          rule: { code: 'leg-nb-max', name: 'Consecutive Flight Duty Periods' },
          message: 'Should be maximum of 3 but currently is 5',
          tag: 'at',
          leg: {
            flightNumber: 7954,
            departureStationCode: 'YYT',
            arrivalStationCode: 'YYR',
            startDate: '2019-07-08',
          },
          contextStartDateTime: '2019-07-08T19:45',
          contextEndDateTime: '2019-07-08T20:35',
        },
      ],
    },
  },
  120: {
    alertLevel: 'info',
    alerts: {
      infoAlerts: [
        {
          rule: { code: 'leg-nb-max', name: 'Consecutive Flight Duty Periods' },
          message: 'Should be maximum of 3 but currently is 5',
          tag: 'at',
          leg: {
            flightNumber: 7951,
            departureStationCode: 'YYT',
            arrivalStationCode: 'YYR',
            startDate: '2019-07-08',
          },
          contextStartDateTime: '2019-07-08T19:45',
          contextEndDateTime: '2019-07-08T20:35',
        },
      ],
    },
  },
  176: {
    alertLevel: 'caution',
    alerts: {
      cautionAlerts: [
        {
          rule: { code: 'leg-nb-max', name: 'Consecutive Flight Duty Periods' },
          message: 'Should be maximum of 3 but currently is 5',
          tag: 'at',
          leg: {
            flightNumber: 7957,
            departureStationCode: 'YYT',
            arrivalStationCode: 'YYR',
            startDate: '2019-07-08',
          },
          contextStartDateTime: '2019-07-08T19:45',
          contextEndDateTime: '2019-07-08T20:35',
        },
      ],
      infoAlerts: [
        {
          rule: { code: 'leg-nb-max', name: 'Consecutive Flight Duty Periods' },
          message: 'Should be maximum of 3 but currently is 5',
          tag: 'at',
          leg: {
            flightNumber: 7965,
            departureStationCode: 'YYT',
            arrivalStationCode: 'YYR',
            startDate: '2019-07-08',
          },
          contextStartDateTime: '2019-07-08T19:45',
          contextEndDateTime: '2019-07-08T20:35',
        },
      ],
    },
  },
};
