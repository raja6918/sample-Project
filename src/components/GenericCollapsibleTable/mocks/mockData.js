import { ellipsisTransformer } from '../../GenericTable/transformers';
import { SelectInput, ColorHighlighter } from '../components';

export const stateData = [
  {
    value: true,
    display: 'On',
  },
  {
    value: false,
    display: 'Off',
  },
];

const handleChange = value => {
  console.log('handle SelectInput Change', value);
};

export const getHeaders = t => [
  {
    field: 'name',
    displayName: t('DATA.rules.headers.name'),
    transformer: ellipsisTransformer,
    transformerStyle: {
      maxWidth: '235px',
    },
  },
  {
    field: 'state',
    displayName: t('DATA.rules.headers.state'),
    disableFilter: true,
    component: SelectInput,
    props: {
      onChange: handleChange,
      items: stateData,
    },
    componentStyle: {
      width: '120px',
      fontSize: '0.8125rem',
    },
    cellStyle: {
      width: '120px',
    },
  },
  {
    field: 'settingsFrom',
    displayName: t('DATA.rules.headers.settingsFrom'),
    component: ColorHighlighter,
    componentStyle: {
      color: '#FF650C',
    },
    props: {
      header: 'isEdited',
    },
  },
  {
    field: 'catagory',
    displayName: t('DATA.rules.headers.catagory'),
  },
  {
    field: 'type',
    displayName: t('DATA.rules.headers.type'),
  },
];

export const rulesData = [
  {
    state: true,
    settingsFrom: 'Baseline Rules',
    catagory: 'CAR Government Regulation',
    name: 'Night Duty',
    type: 'Definition',
    id: 1,
  },
  {
    state: false,
    settingsFrom: 'Flight Desk',
    catagory: 'CAR Government Regulation',
    name:
      'Split Duty Extension to Maximum FDP (Split Duty Extension to Maximum FDP)',
    type: 'Duty Time Limitation',
    id: 2,
  },
  {
    state: true,
    settingsFrom: 'Baseline Rules',
    catagory: 'CAR Government Regulation',
    name: 'Minimum Rest at Home Base (Minimum Rest at Home Base)',
    type: 'Rest RUle',
    id: 3,
  },
  {
    isEdited: true,
    state: false,
    settingsFrom: 'Widebody',
    catagory: 'OverSeas Cabin Contract',
    name: 'Earily Duty',
    type: 'Cost RUle',
    id: 4,
  },
  {
    state: true,
    settingsFrom: 'Baseline Rules',
    catagory: 'Basic Pairing Rules',
    name: 'Late Duty',
    type: 'Soft Rule',
    id: 5,
  },
];
