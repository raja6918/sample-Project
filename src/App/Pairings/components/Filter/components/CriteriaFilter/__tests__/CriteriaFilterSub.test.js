import React from 'react';
import { shallow } from 'enzyme';
import CriteriaFilterSub from '../CriteriaFilterSub';
import { filterCriteria } from '../../../mockData';

describe('Test CriteriaFilterSub Component', () => {
  const minProps = {
    t: msg => msg,
    id: 1,
    subMenuOpen: 2,
    catElement: filterCriteria[0].categories[0],
    getfilteredCriteria: jest.fn(),
    handleClose: jest.fn(),
    handleSubMenuOpen: jest.fn(),
    handleSubMenuClose: jest.fn(),
  };

  let wrapper;
  beforeAll(() => {
    wrapper = shallow(<CriteriaFilterSub {...minProps} />);
  });

  test('CriteriaFilterSub Component renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('toggleItems works correctly on checked', () => {
    const items = [
      { name: 'item1', crId: 1 },
      { name: 'item2', crId: 2 },
      { name: 'item3', crId: 3 },
    ];
    const e = { target: { checked: true } };
    wrapper.instance().setState({
      filteredItems: [items[0], items[1]],
      selectedIds: [items[0].crId, items[1].crId],
    });
    const state = {
      filteredItems: wrapper.state('filteredItems'),
      selectedIds: wrapper.state('selectedIds'),
    };
    wrapper.instance().toggleItems(items[2], e.target.checked);
    expect(wrapper.state('filteredItems')).toEqual([
      ...state.filteredItems,
      items[2],
    ]);
    expect(wrapper.state('selectedIds')).toEqual([
      ...state.selectedIds,
      items[2].crId,
    ]);
  });

  test('toggleItems works correctly on unchecked', () => {
    const items = [
      { name: 'item1', crId: 1 },
      { name: 'item2', crId: 2 },
      { name: 'item3', crId: 3 },
    ];
    const e = { target: { checked: false } };
    wrapper.instance().setState({
      filteredItems: items,
      selectedIds: items.map(elem => elem.crId),
    });
    wrapper.instance().toggleItems(items[2], e.target.checked);
    expect(wrapper.state('filteredItems')).toEqual(
      items.filter(elem => elem.crId !== items[2].crId)
    );
    expect(wrapper.state('selectedIds')).toEqual(
      items.map(elem => elem.crId).filter(elem => elem !== items[2].crId)
    );
  });

  test('handleFilter works correctly', () => {
    const e = { target: { value: 'Filtering' } };
    wrapper.instance().handleFilter(e);
    expect(wrapper.state('filter')).toBe('Filtering');
  });

  test('handleClearFilter works correctly', () => {
    wrapper.instance().handleClearFilter();
    expect(wrapper.state('filter')).toBe('');
  });

  test('handleClearAll works correctly', () => {
    wrapper
      .instance()
      .setState({ filteredItems: ['item1', 'item2'], selectedIds: [1, 2] });
    wrapper.instance().handleClearAll();
    expect(wrapper.state('filteredItems')).toEqual([]);
    expect(wrapper.state('selectedIds')).toEqual([]);
  });
});
