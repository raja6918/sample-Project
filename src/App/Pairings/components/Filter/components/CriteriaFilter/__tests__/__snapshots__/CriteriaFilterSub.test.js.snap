// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test CriteriaFilterSub Component CriteriaFilterSub Component renders correctly 1`] = `
<div>
  <WithStyles(WithStyles(ForwardRef(MenuItem)))
    id="main-menu"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    selected={false}
  >
    <WithStyles(ForwardRef(ListItemText))
      primary="FILTER.filterCriteria.categories.TRIP_PAIRING_CRITERIA"
    />
    <WithStyles(ForwardRef(ListItemIcon))
      style={
        Object {
          "color": "#3f3e3e",
          "minWidth": "20px",
        }
      }
    >
      <Memo([object Object])
        fontSize="small"
      />
    </WithStyles(ForwardRef(ListItemIcon))>
  </WithStyles(WithStyles(ForwardRef(MenuItem)))>
  <WithStyles(Component)
    anchorEl={null}
    id="customized-menu"
    keepMounted={true}
    open={false}
    style={
      Object {
        "pointerEvents": "none",
      }
    }
  >
    <CriteriaFilterSub__StyledSubComponent
      className="sub-menu"
      popupHeight={0}
      style={
        Object {
          "pointerEvents": "auto",
        }
      }
    >
      <div>
        <CriteriaFilterSub__StyledInputSearch>
          <WithStyles(ForwardRef(Input))
            disableUnderline={true}
            endAdornment=""
            onChange={[Function]}
            onKeyDown={[Function]}
            placeholder="SEARCHENGINE"
            startAdornment={
              <FilterInput__SearchIcon>
                search
              </FilterInput__SearchIcon>
            }
            type="text"
            value=""
          />
        </CriteriaFilterSub__StyledInputSearch>
      </div>
      <ScrollBar
        className=""
        component="div"
        containerRef={[Function]}
        option={
          Object {
            "swipeEasing": true,
            "wheelSpeed": 0.4,
          }
        }
      >
        <div>
          <div
            key="110"
            onClick={[Function]}
            style={
              Object {
                "background": "",
                "fontWeight": "",
              }
            }
          >
            <WithStyles(ForwardRef(Checkbox))
              checked={false}
            />
            <div>
              FILTER.filterCriteria.criteria.TRIP_PAIRING_CR_BLOCK_TIME
            </div>
          </div>
          <div
            key="115"
            onClick={[Function]}
            style={
              Object {
                "background": "",
                "fontWeight": "",
              }
            }
          >
            <WithStyles(ForwardRef(Checkbox))
              checked={false}
            />
            <div>
              FILTER.filterCriteria.criteria.TRIP_PAIRING_CR_BPDH
            </div>
          </div>
          <div
            key="116"
            onClick={[Function]}
            style={
              Object {
                "background": "",
                "fontWeight": "",
              }
            }
          >
            <WithStyles(ForwardRef(Checkbox))
              checked={false}
            />
            <div>
              FILTER.filterCriteria.criteria.TRIP_PAIRING_CR_BPDP
            </div>
          </div>
          <div
            key="19"
            onClick={[Function]}
            style={
              Object {
                "background": "",
                "fontWeight": "",
              }
            }
          >
            <WithStyles(ForwardRef(Checkbox))
              checked={false}
            />
            <div>
              FILTER.filterCriteria.criteria.TRIP_PAIRING_CR_CC
            </div>
          </div>
          <div
            key="17"
            onClick={[Function]}
            style={
              Object {
                "background": "",
                "fontWeight": "",
              }
            }
          >
            <WithStyles(ForwardRef(Checkbox))
              checked={false}
            />
            <div>
              FILTER.filterCriteria.criteria.TRIP_PAIRING_CR_CNX_ACHG
            </div>
          </div>
          <div
            key="15"
            onClick={[Function]}
            style={
              Object {
                "background": "",
                "fontWeight": "",
              }
            }
          >
            <WithStyles(ForwardRef(Checkbox))
              checked={false}
            />
            <div>
              FILTER.filterCriteria.criteria.TRIP_PAIRING_CR_CNX_AT
            </div>
          </div>
          <div
            key="16"
            onClick={[Function]}
            style={
              Object {
                "background": "",
                "fontWeight": "",
              }
            }
          >
            <WithStyles(ForwardRef(Checkbox))
              checked={false}
            />
            <div>
              FILTER.filterCriteria.criteria.TRIP_PAIRING_CR_CNX_DURATION
            </div>
          </div>
          <div
            key="18"
            onClick={[Function]}
            style={
              Object {
                "background": "",
                "fontWeight": "",
              }
            }
          >
            <WithStyles(ForwardRef(Checkbox))
              checked={false}
            />
            <div>
              FILTER.filterCriteria.criteria.TRIP_PAIRING_CR_CREW_BASE
            </div>
          </div>
          <div
            key="113"
            onClick={[Function]}
            style={
              Object {
                "background": "",
                "fontWeight": "",
              }
            }
          >
            <WithStyles(ForwardRef(Checkbox))
              checked={false}
            />
            <div>
              FILTER.filterCriteria.criteria.TRIP_PAIRING_CR_DUTIES
            </div>
          </div>
          <div
            key="114"
            onClick={[Function]}
            style={
              Object {
                "background": "",
                "fontWeight": "",
              }
            }
          >
            <WithStyles(ForwardRef(Checkbox))
              checked={false}
            />
            <div>
              FILTER.filterCriteria.criteria.TRIP_PAIRING_CR_DUTY_DAYS
            </div>
          </div>
          <div
            key="12"
            onClick={[Function]}
            style={
              Object {
                "background": "",
                "fontWeight": "",
              }
            }
          >
            <WithStyles(ForwardRef(Checkbox))
              checked={false}
            />
            <div>
              FILTER.filterCriteria.criteria.TRIP_PAIRING_CR_LAYOVER_AT
            </div>
          </div>
          <div
            key="118"
            onClick={[Function]}
            style={
              Object {
                "background": "",
                "fontWeight": "",
              }
            }
          >
            <WithStyles(ForwardRef(Checkbox))
              checked={false}
            />
            <div>
              FILTER.filterCriteria.criteria.TRIP_PAIRING_CR_LAYOVER_DUR
            </div>
          </div>
          <div
            key="112"
            onClick={[Function]}
            style={
              Object {
                "background": "",
                "fontWeight": "",
              }
            }
          >
            <WithStyles(ForwardRef(Checkbox))
              checked={false}
            />
            <div>
              FILTER.filterCriteria.criteria.TRIP_PAIRING_CR_OCCURRENCES
            </div>
          </div>
          <div
            key="117"
            onClick={[Function]}
            style={
              Object {
                "background": "",
                "fontWeight": "",
              }
            }
          >
            <WithStyles(ForwardRef(Checkbox))
              checked={false}
            />
            <div>
              FILTER.filterCriteria.criteria.TRIP_PAIRING_CR_OP_FLIGHTS
            </div>
          </div>
          <div
            key="14"
            onClick={[Function]}
            style={
              Object {
                "background": "",
                "fontWeight": "",
              }
            }
          >
            <WithStyles(ForwardRef(Checkbox))
              checked={false}
            />
            <div>
              FILTER.filterCriteria.criteria.TRIP_PAIRING_CR_PAIRING_END
            </div>
          </div>
          <div
            key="120"
            onClick={[Function]}
            style={
              Object {
                "background": "",
                "fontWeight": "",
              }
            }
          >
            <WithStyles(ForwardRef(Checkbox))
              checked={false}
            />
            <div>
              FILTER.filterCriteria.criteria.TRIP_PAIRING_CR_PAIRING_END_TIME
            </div>
          </div>
          <div
            key="11"
            onClick={[Function]}
            style={
              Object {
                "background": "",
                "fontWeight": "",
              }
            }
          >
            <WithStyles(ForwardRef(Checkbox))
              checked={false}
            />
            <div>
              FILTER.filterCriteria.criteria.TRIP_PAIRING_CR_PAIRING_NAME
            </div>
          </div>
          <div
            key="13"
            onClick={[Function]}
            style={
              Object {
                "background": "",
                "fontWeight": "",
              }
            }
          >
            <WithStyles(ForwardRef(Checkbox))
              checked={false}
            />
            <div>
              FILTER.filterCriteria.criteria.TRIP_PAIRING_CR_PAIRING_START
            </div>
          </div>
          <div
            key="119"
            onClick={[Function]}
            style={
              Object {
                "background": "",
                "fontWeight": "",
              }
            }
          >
            <WithStyles(ForwardRef(Checkbox))
              checked={false}
            />
            <div>
              FILTER.filterCriteria.criteria.TRIP_PAIRING_CR_PAIRING_START_TIME
            </div>
          </div>
          <div
            key="111"
            onClick={[Function]}
            style={
              Object {
                "background": "",
                "fontWeight": "",
              }
            }
          >
            <WithStyles(ForwardRef(Checkbox))
              checked={false}
            />
            <div>
              FILTER.filterCriteria.criteria.TRIP_PAIRING_CR_TAFB
            </div>
          </div>
        </div>
      </ScrollBar>
      <div>
        <CriteriaFilterSub__AddButton
          color="primary"
          disabled={true}
          onClick={[Function]}
          variant="contained"
        >
          FILTER.pane.buttons.add
        </CriteriaFilterSub__AddButton>
        <CriteriaFilterSub__ClearAllButton
          color="primary"
          disableFocusRipple={true}
          disableRipple={true}
          disabled={true}
          onClick={[Function]}
          size="small"
          startIcon={<Memo />}
        >
          FILTER.pane.buttons.clearAll
        </CriteriaFilterSub__ClearAllButton>
      </div>
    </CriteriaFilterSub__StyledSubComponent>
  </WithStyles(Component)>
</div>
`;
